actionpack/lib/action_controller/metal/renderers.rb

```diff
@@ -65,19 +98,49 @@ def use_renderers(*args)
         self._renderers = renderers.freeze
       end
       alias use_renderer use_renderers
+
+      # See <tt>Renderers.use_renderers</tt>
+      def use_serializers(*args)
+        serializers = _serializers + args
+        self._serializers = serializers.freeze
+      end
+      alias use_serializer use_serializers
+    end
+
+    def has_serializer?(key)
+      _serializers.key?(key)
     end
 
+    def serializer_for(key)
+      _serializers[key]
+    end
+
+    def assert_serializer!(serializer_name)
+      return true if has_serializer?(serializer_name)
+      raise ActionController::MissingSerializer, "There is no '#{serializer_name}' serializer.\n" <<
+            "Known serializers are #{_serializers.keys}"
+    end
```

Comment:
Only that it makes the `_render_to_body_with_renderer` method a little more manageable

```diff
    def _render_to_body_with_renderer(options)
      _renderers.each do |name|
-        if options.key?(name)
-          _process_options(options)
-          method_name = Renderers._render_with_renderer_method_name(name)
-          return send(method_name, options.delete(name), options)
-        end
+        next unless options.key?(name)
+        _process_options(options)
+        method_name = Renderers._render_with_renderer_method_name(name)
+        value = options.delete(name)
+        serializer_name = options.key?(:serializer_name) ? options.delete(:serializer_name) : name
+        assert_serializer!(serializer_name)
+        serializer = serializer_for(serializer_name)
+        value = serializer.call(value, options)
+        return send(method_name, value, options)
      end
      nil
    end
```